rest/upgradetest/upgrade_registry_test.go

```diff
@@ -190,26 +185,14 @@ func TestLegacyMetadataID(t *testing.T) {
 	assert.NoError(t, err)
 	legacyRT.Close()
 
-	persistentRT := rest.NewRestTester(t, &rest.RestTesterConfig{
+	// use DefaultCollection since we're testing legacy behavior
+	persistentRT := rest.NewRestTesterDefaultCollection(t, &rest.RestTesterConfig{
 		CustomTestBucket: tb1,
 		PersistentConfig: true,
 	})
 	defer persistentRT.Close()
 
-	// Generate a dbConfig from the legacy startup config using ToStartupConfig, and use it to create a database
-	_, dbMap, err := legacyConfig.ToStartupConfig(testCtx)
-	require.NoError(t, err)
-
-	dbConfig, ok := dbMap["db"]
-	require.True(t, ok)
-
-	// Need to sanitize the db config, but can't use sanitizeDbConfigs because it assumes non-empty server address
-	dbConfig.Username = ""
-	dbConfig.Password = ""
-	dbConfigBytes, err := base.JSONMarshal(dbConfig)
-
-	require.NoError(t, err)
-	resp = persistentRT.SendAdminRequest("PUT", "/db/", string(dbConfigBytes))
+	resp = persistentRT.CreateDatabase("db", persistentRT.NewDbConfig())
```

Comment:
I think the use of ToStartupConfig might be intentional in this test, as it's trying to cover metadataID handling a legacy config is being upgraded.  It feels like using NewDbConfig() might hide potential bugs in the upgrade process?